### PR TITLE
[120] Implement actor's resume deletion API @DELETE

### PR DESCRIPTION
**Pull request by SuitedIceZ**
## What did you do
- edit database schema : change actor in resume table to optional
- Implement ```DELETE resumes/:resumeId``` for actors to delete their own resumes
- There are 2 deleting case 
  1. the resume have application -> remove actor from that resume and remove that resume from actor
  2. the resume not have application -> remove that resume from database
- Write unit test for ```DELETE resumes/:resumeId``` in resume.service.spec.ts (only for case ii.)

## Demo
- https://api-dev.modela.miello.dev/swagger#/auth/AuthController_login
- login with (or any actor and any roles)
```
{
  "email": "actor2@gmail.com",
  "password": "password"
}
```
- https://api-dev.modela.miello.dev/swagger#/resumes/ResumeController_getResumes
- get all actor's resumes
- https://api-dev.modela.miello.dev/swagger#/resumes/ResumeController_createResume
- create new resume if none
```
{
  "name": "very cute cat",
  "resumeUrl": "cat.png"
}
```
- https://api-dev.modela.miello.dev/swagger#/resumes/ResumeController_deleteResume
- try delete any resumes with resumeId from get
- delete the seeded resumes is more likely to already have application
- also try with different roles and different actor owners

### CODE REVIEW

File: apps/api/src/modules/resume/resume.controller.ts
Comment (SuitedIceZ) [reply apps/api/src/modules/resume/resume.controller.ts:64]: it's should respond with a 204 status and an empty body???

**Comment by SuitedIceZ** [comment]
> Oops, เหมือนจะเจอปัญหาเกี่ยวกับ foreign key อันที่มี application , อาบน้ำเสร็จเดี๋ยวมาแก้ 🙏

แก้แล้วเรียบร้อย

**Review by miello — COMMENTED** [review]
LGTM, please wait for review from @palmcm

File: apps/api/src/modules/resume/resume.controller.ts
Comment (palmcm) [inline apps/api/src/modules/resume/resume.controller.ts:74]:
```suggestion
  @ApiNoContentResponse({ description: 'Delete resume complete.' })
```
use @ApiNoContentResponse instead

**Review by palmcm — APPROVED** [review]
Please Change decorator first then nothing more

File: apps/api/src/modules/resume/resume.respository.ts
Comment (miello) [inline apps/api/src/modules/resume/resume.respository.ts:61]: Does resumeId already in number ? please remove unnecessary type casting

**Review by miello — APPROVED** [review]
Some redundant type casting

**Pull request merged**
